runbotics/runbotics-orchestrator-ui/src/src-app/utils/APIResource.ts

```diff
@@ -0,0 +1,114 @@
+                const { auth: { user } } = thunkApi.getState();
+
+                const url = this.buildURL({ resourcePath, resourceId, pageParams }, user);
+
+                return axios.get<ReturnType>(url, payload)
+                    .then(response => response.data)
+                    .catch(error => thunkApi.rejectWithValue(error.response.data));
+            }
+        );
+    }
+
+    static post<ReturnType, PayloadType>(
+        typePrefix: string, resourcePath?: string
+    ): AsyncThunk<ReturnType, PayloadWrap<PayloadType>, { state: RootState }> {
+        return createAsyncThunk<ReturnType, PayloadWrap<PayloadType>, { state: RootState }>(
+            typePrefix,
+            ({ payload, resourceId }, thunkApi) => {
+                const { auth: { user } } = thunkApi.getState();
+
+                const url = this.buildURL({ resourcePath, resourceId }, user);
+
+                return axios.post<ReturnType>(url, payload)
+                    .then(response => response.data)
+                    .catch(error => thunkApi.rejectWithValue(error.response.data));
+            }
+        );
+    }
+
+    static patch<ReturnType, PayloadType>(
+        typePrefix: string, resourcePath?: string
+    ): AsyncThunk<ReturnType, PayloadWrap<PayloadType>, { state: RootState }> {
+        return createAsyncThunk<ReturnType, PayloadWrap<PayloadType>, { state: RootState }>(
+            typePrefix,
+            ({ payload, resourceId }, thunkApi) => {
+                const { auth: { user } } = thunkApi.getState();
+
+                const url = this.buildURL({ resourcePath, resourceId }, user);
+
+                return axios.patch<ReturnType>(url, payload)
+                    .then(response => response.data)
+                    .catch(error => thunkApi.rejectWithValue(error.response.data));
+            }
+        );
+    }
+
+    static delete<ReturnType, PayloadType>(
+        typePrefix: string, resourcePath?: string
+    ): AsyncThunk<ReturnType, PayloadWrap<PayloadType>, { state: RootState }> {
+        return createAsyncThunk<ReturnType, PayloadWrap<PayloadType>, { state: RootState }>(
+            typePrefix,
+            ({ payload, resourceId }, thunkApi) => {
+                const { auth: { user } } = thunkApi.getState();
+
+                const url = this.buildURL({ resourcePath, resourceId }, user);
+
+                return axios.delete<ReturnType>(url, payload)
+                    .then(response => response.data)
+                    .catch(error => thunkApi.rejectWithValue(error.response.data));
+            }
+        );
+    }
+
+    private static buildURL(
+        { resourcePath, resourceId, pageParams }: PathElements,
+        user: User
+    ) {
+        const resourcePathPart = resourcePath
+            ? `/${resourcePath}` : '';
+
+        const resourceIdPart = resourceId
+            ? `/${resourceId}` : '';
+
+        const apiURL = hasRoleAccess(user, [Role.ROLE_ADMIN])
```

Comment:
admin should be able to call all endpoints, it's business logic if we fetch for him only his resources or something more and it shouldn't be decided in a util